src/zarr/abc/codec.py

```diff
@@ -16,6 +25,40 @@
     from zarr.metadata import ArrayMetadata
     from zarr.config import RuntimeConfiguration
 
+T = TypeVar("T")
+U = TypeVar("U")
+
+
+def noop_for_none(
+    func: Callable[[T, ArraySpec, RuntimeConfiguration], Awaitable[Optional[U]]],
```

Comment:
```suggestion
    func: Callable[[T, ArraySpec, RuntimeConfiguration], Awaitable[U | None]],
```

Reply:
Can we do these changes automated?

Reply:
pylint has https://pylint.readthedocs.io/en/stable/user_guide/messages/refactor/consider-alternative-union-syntax.html

Reply:
i'm not aware of an automated way to do it, but it would be a lot nicer than the manual way. you can also do an IDE-driven style where you remove the `Union` import from typing and find all the errors that result.